static/js/lib/selectors.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import { createSelector } from "reselect"
+import { propOr } from "ramda"
+import qs from "query-string"
+
+export const qsSelector = createSelector(
+  (_, ownProps) => ownProps.location.search,
+  qs.parse
+)
```

Comment:
I think instead of using this selector you could just lean on `react-routers`'s `useParams` hook. you should be able to do something like

```js
const { verification_code } = useParams()
```

in all your components

Reply:
https://reacttraining.com/react-router/web/api/Hooks/useparams

Reply:
@aliceriot @pdpinch should the code be refactored according to these suggestions now or later in a subsequent PR?  And should the same changes be made to the matching files in mitxpro?

Reply:
I didn't realize that all of this was just copied over code, I thought maybe just some of the library functions and so on were. I think it should be changed at some point but it doesn't need to be now. Maybe we should just file an issue for it? There are definitely things that I think should be cleaned up, like the `<select>` components, some code re-use improvements for the validation, and so on, but it's not necessary to do it to merge I think.